### PR TITLE
task: clean up mongosh package

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Did a quick pass when refactoring some code for #652 and these couple of thing came up

- the name of the validation was probably a copy/paste, fixed and moved to our `validate` package
- some function names could be simplified 
- some functions could be split to facilitate testing
- added some testing

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/mongosh/mongosh_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package mongosh
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestBin(t *testing.T) {
+	want := mongoshBin
+	if isWindows() {
```
Comment (gssbzn) [inline internal/mongosh/mongosh_test.go:10]: since the github actions run on both linux and windows this should be a good enough test

**Review by threebee — DISMISSED** [review]
LGTM

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

File: internal/mongosh/mongosh.go [hunk 2]
```diff
@@ -43,24 +47,5 @@ func FindBinaryInPath() string {
...
-	err := syscall.Exec(binary, args, env)
-
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func ValidateUniqueUsername(val interface{}) error {
```
Comment (andreaangiolillo) [inline internal/mongosh/mongosh.go:55]: ops 🤦

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**